carmastate_mod.F90

```diff
@@ -175,13 +161,6 @@ subroutine CARMASTATE_Create(cstate, carma_ptr, time, dtime, NZ, igridv, igridh,
     ! Calculate the metrics, ...
     ! if Cartesian coordinates were specifed, then the units need to be converted
     ! from MKS to CGS.
-    if (cstate%f_igridh == I_CART) then
```

Comment:
Perhaps what we should do is get rid of lat/lon and make xc and yc either the center of the grid in lat/lon coordinates or in cartesian coordinates. I think the only place these are used are in error messages to indicate where the problem happened. So maybe they turn into optional fields and are indicates as arbitrary x and y locations only used for reporting? There are places in the model carma_model_mod.F90 for specific models wher ewe do use lat/lon. So perhaps we leave all 4, but make them all optional fields for CARMA, but always specify just the lat/lon in the CAM/CARMA case.